erpnext/accounts/doctype/payment_entry/payment_entry.js

```diff
@@ -800,22 +800,18 @@ frappe.ui.form.on('Payment Entry', {
 				//If allocate payment amount checkbox is unchecked, set zero to allocate amount
 				row.allocated_amount = 0;
 
-			} else if (frappe.flags.allocate_payment_amount != 0 && !row.allocated_amount) {
+			} else if (frappe.flags.allocate_payment_amount != 0 && (!row.allocated_amount || paid_amount_change)) {
 				if (row.outstanding_amount > 0 && allocated_positive_outstanding > 0) {
```

Comment:
```suggestion
				if (row.outstanding_amount > 0 && allocated_positive_outstanding >= 0) {
```
You won't need the second else if condition then